central/image/service/service_impl.go

```diff
@@ -440,10 +440,11 @@ func (s *serviceImpl) EnrichLocalImageInternal(ctx context.Context, request *v1.
 		Id:   imgID,
 		Name: request.GetImageName(),
 		// 'Names' must be populated to enable cache hits in central AND sensor.
-		Names:          []*storage.ImageName{request.GetImageName()},
+		Names:          buildNames(request.GetImageName(), request.GetMetadata()),
 		Signature:      request.GetImageSignature(),
 		Metadata:       request.GetMetadata(),
 		Notes:          request.GetImageNotes(),
+		Scan:           existingImg.GetScan(),
```

Comment:
why wasn't this set before?

Reply:
My guess is it wasn't intentional and was a bug - was causing 0 components to be returned to Sensor under various conditions, such as:

If sensor is bounced (cache cleared) - then Sensor calls `EnrichLocalImageInternal` for each image that is cluster local - If `forceScanUpdate` is `false` (meaning scan in central DB is new within the last 4 hours) but `forceSigVerificationUpdate` is `true` (which in my testing is almost always the case), no scan data was returned to Sensor. 

I'm assuming this would break policy - I noticed it when delegating scans via `roxctl` and realizing the 'first' scan would give me components / vulnerabilities, but the second scan would not (if not using `force`)